biblib/tests/unit_tests/test_manage.py

```diff
@@ -197,5 +176,300 @@ def test_delete_stale_users(self):
+                    permissions = session.query(Permissions).filter(Permissions.user_id == service_user).all()
+                    libraries = [session.query(Library).filter(Library.id == permission.library_id).one() for permission in permissions if permission.permissions['owner']]
+                    LibraryVersion = sqlalchemy_continuum.version_class(Library)
+                    revision_lengths = []
+                    for library in libraries:
+                        revisions = session.query(LibraryVersion).filter_by(id=library.id).all()
+                        revision_lengths.append(len(revisions))
+                
+                # Now run the obsolete deletion
+                DeleteObsoleteVersionsNumber().run(app=self.app, n_revisions=self.n_revisions)
+                service_user = user_1_id
+                permissions = session.query(Permissions).filter(Permissions.user_id == service_user).all()
+                libraries = [session.query(Library).filter(Library.id == permission.library_id).one() for permission in permissions if permission.permissions['owner']]
+                LibraryVersion = sqlalchemy_continuum.version_class(Library)
+                updated_revision_lengths = []
+                
+                #confirm most recent remaining revision matches current state of library
+                for library in libraries:
+                    updated_revisions = session.query(LibraryVersion).filter_by(id=library.id).all()
+                    updated_revision_lengths.append(len(updated_revisions))
+                    self.assertUnsortedEqual(library.bibcode, updated_revisions[-1].bibcode) 
+                
+                #Confirm number of revisions matches expected length
+                for i in range(0,len(updated_revision_lengths)):
+                    self.assertEqual(revision_lengths[i]-updated_revision_lengths[i], revision_lengths[i]-self.n_revisions)
+    
+
+            except Exception:
+                raise
+            finally:
+                # Destroy the tables
+                session.execute('drop table users;')
+                pass
+
+    def test_delete_obsolete_versions_time(self):
+        """
+        Tests that the DeleteStaleUsers action that propogates the deletion of
+        users from the API database to that of the microservice.
+
+        :return: no return
+        """
+
+        with self.app.session_scope() as session:
+            # We do not add user 1 to the API database
+            session.execute('create table users (id integer, random integer);')
+            session.execute('insert into users (id, random) values (2, 7);')
+            session.commit()
+
+        with self.app.session_scope() as session:
+            try:
+
+                # Add some content to the users, libraries, and permissions within
+                # the microservices
+                user_1 = User(absolute_uid=1)
+                session.add(user_1)
+                session.commit()
```

Comment:
And check if user_1 needs to be added twice (here and on line 347) or not

Reply:
Same as above.